test/peer_test.dart

```diff
@@ -184,4 +184,23 @@ void main() {
       incoming.add({"completely": "wrong"});
     });
   });
+
+  test("can notify on unhandled errors for if the method throws", () async {
+    Exception exception = Exception('test exception');
+    var incomingController = new StreamController();
+    var outgoingController = new StreamController();
+    final Completer<Exception> completer = Completer<Exception>();
+    peer = new json_rpc.Peer.withoutJson(
+        new StreamChannel(incomingController.stream, outgoingController),
+        onUnhandledError: (error, stack) {
+      completer.complete(error);
```

Comment:
Nit: need a little more indentation (`completer.complete` should be inside and more indented than `onUnhandledError: `).

Reply:
That's how dartfmt did it

Reply:
Add a comma between `}` and `);` below and reformat...

Reply:
Done

Reply:
Ah, this is magic. Is the old indentation a feature or an unexpected behavior of dartfmt?

Reply:
this is an...uh...obscure feature of dartfmt 😄